packages/augur-sdk/src/state/db/MarketDB.ts

```diff
@@ -296,15 +297,20 @@ export class MarketDB extends DerivedDB {
 
   processNewBlock = async (block: Block): Promise<void> => {
     const timestamp = (await this.augur.getTimestamp()).toNumber();
-    await this.processTimestamp(timestamp, parseInt(block.number))
-  }
+    await this.processTimestamp(timestamp, Number(block.number))
+  };
 
   processTimestampSet = async (log: TimestampSetLog): Promise<void> => {
     const timestamp = new BigNumber(log.newTimestamp).toNumber();
     await this.processTimestamp(timestamp, log.blockNumber)
-  }
+  };
 
   private async processTimestamp(timestamp: number, blockNumber: number): Promise<void> {
+
+    while (this.locks['processDisputeCrowdsourcerCompleted']) {
+      await sleep(50);
```

Comment:
prob need ability to just break out and die. We don't want to sleep forever. I suggest putting a setTimeout or some way to break out after 1 second

Reply:
Since I'm adding this logic to DerivedDB, I'll put the helper function there to do that.